### ISSUE TITLE
Multiple problems with ender dragon entity 

### ISSUE THREAD

**Issue by NeumimTo** [SpongePowered/Sponge#1651]
**I am currently running**

- SpongeVanilla version: 1.12.2-7.0.0-BETA-372
- Java version: 8
- Operating System: linux

- Plugins/Mods: none 
  
1) problem (no longer relevant see comments)
  - Interact with block
  - Entity is spawned on the server side but client receives no packet
  - to see the dragon player have to re-login. 

 2) problem 
   - player is kicked after dismounting (pressing shit) and cannot login back. 

After dismounting:

```
[21:06:09 INFO]: NeumimTo lost connection: Invalid move player packet received
[21:06:09 INFO]: NeumimTo left the game
[21:06:09 WARN]: handleDisconnection() called twice

```
   
After reloging server kicks me with following log:

```
[20:58:20 INFO]: NeumimTo joined the game
[20:58:21 INFO]: NeumimTo lost connection: Invalid move player packet received
[20:58:21 INFO]: NeumimTo left the game
```
3. Methods Entity#getRotation() and Entity#setRotation(Vec3d) wont affect the entity in a correct way ,
    after setting the entity rotation as shown below the entity jaw is always player's jaw + 180 deg.

```Java

    @Listener
    public void start(InteractBlockEvent event, @Root Player player) {
Entity dog = player.getWorld().createEntity(EntityTypes.ENDER_DRAGON, player.getLocation().getPosition());
        player.getWorld().spawnEntity(dog);

        Sponge.getScheduler().createTaskBuilder().execute(() ->
                dog.addPassenger(player)
 
               Sponge.getScheduler().createTaskBuilder()
                    .interval(1, TimeUnit.SECONDS)
                    .execute(() -> {
                        List<Entity> passengers = dog.getPassengers();
                        if (passengers.isEmpty())
                            return;
                        Entity entity1 = passengers.get(0);
                        Vector3d rotation = entity1.getRotation();
                        dog.setRotation(rotation);
                    })
                    .submit(this);
        ).submit(this);
    }

    @Listener
    public void blockEntityDamage(DamageEntityEvent event, @Root EnderDragon enderDragon) {
        event.setCancelled(true);
    }

    @Listener
    public void blockWorldDamage(ChangeBlockEvent event, @Root EnderDragon enderDragon) {
        event.setCancelled(true);
    }


```

**Comment by Eufranio** [comment]
Try to spawn the entity via `world.spawnEntity(Entity e)`, from [EntityUniverse](https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/world/extent/EntityUniverse.java#L259), before adding it's passenger.

**Comment by NeumimTo** [comment]
That helped with the first problem, but dismounting remains broken

**Comment by JBYoshi** [comment]
I've traced it down to a "divide by zero" error in `EntityDragon`.

**Comment by JBYoshi** [comment]
Fixed in 01f6126d4b8229ca019b4f81293320d768ca6dc3.
